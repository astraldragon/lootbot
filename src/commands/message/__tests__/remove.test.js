jest.mock("discord-akairo");

jest.mock("../../../models", () => ({
  Message: { destroy: jest.fn("destroy") }
}));

const { Message } = require("../../../models");
const { Command } = require("discord-akairo");
const { checkManagePermissions } = require("../../../support");
const RemoveCommand = require("../remove");

describe("commands/message/remove", () => {
  beforeEach(() => {
    this.args = { name: "msg1" };
    this.command = new RemoveCommand();

    this.msg = {
      guild: { id: "1234" },
      channel: {
        send: jest.fn("send").mockReturnValue(Promise.resolve())
      }
    };
  });

  it("should configure properly", () => {
    expect(Command).toMatchSnapshot();
  });

  it("should have the manage permission check", () => {
    expect(Command.mock.calls[0][1].options.permissions).toBe(
      checkManagePermissions
    );
  });

  it("should destroy the passed in message", async () => {
    Message.destroy.mockReturnValue(1);
    await this.command.exec(this.msg, this.args);

    expect(Message.destroy).toHaveBeenCalledWith({
      where: { name: this.args.name, guild: this.msg.guild.id }
    });

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `${this.args.name} removed`
    );
  });

  it("should notify when the passed in message does not exist", async () => {
    Message.destroy.mockReturnValue(0);
    await this.command.exec(this.msg, this.args);

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `${this.args.name} not found`
    );
  });

  it("should notify if any errors occur", async () => {
    Message.destroy.mockImplementation(() => {
      throw new Error("test");
    });

    await this.command.exec(this.msg, this.args);

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `An error occurred removing ${this.args.name}`
    );
  });
});
