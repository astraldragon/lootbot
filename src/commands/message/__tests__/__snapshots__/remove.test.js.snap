// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands/message/remove should configure properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "message-remove",
      Object {
        "aliases": Array [
          "message-remove",
          "mr",
        ],
        "args": Array [
          Object {
            "id": "name",
            "prompt": Object {
              "prompt": "What is the name of the message?",
            },
            "type": "string",
          },
        ],
        "category": "Message",
        "channelRestriction": "guild",
        "description": Object {
          "content": "Remove a message",
          "examples": Array [
            "message-remove msg1",
          ],
        },
        "options": Object {
          "permissions": [Function],
        },
      },
    ],
  ],
}
`;
